### ISSUE TITLE
`NewType` no longer compatible with `type` parameter in 1.1.402

### ISSUE THREAD

**Issue by brearleyt** [microsoft/pyright#10598]
**Describe the bug**

In Pyright 1.1.401, passing a `NewType` to a parameter typed as `type` worked without issue. However, in Pyright 1.1.402, this causes a type error .

**Code or Screenshots**

```python
from typing import NewType, reveal_type

Foo = NewType("Foo", int)
reveal_type(Foo)


def func(foo: type) -> int:
    return 0


func(Foo)
```

Output:

```
Type of "Foo" is "type[Foo]"

Argument of type "type[Foo]" cannot be assigned to parameter "foo" of type "type" in function "func"
  "FunctionType" is not assignable to "type"  (reportArgumentType)
```

[Playground link](https://pyright-play.net/?pyrightVersion=1.1.402&pythonVersion=3.13&pythonPlatform=Darwin&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByApgO4AqiRANFCEQG5ECGANgPrwJEBQ3AYmDBQAvIVIUuACgBEAsNJqoYASm51GrDpUlzVvACZFgUYAFcUAY0nBBALliVlUALQA%2BTChi3uUX7SIwpiAoUAAMvNxmljqCqkA).

**VS Code extension or command-line**
pyright 1.1.402

**Comment by erictraut** [comment]
This change was intentional. See https://github.com/microsoft/pyright/discussions/10596 for details.

**Comment by jorenham** [comment]
Note that `reveal_type(Foo)` is still reported as `Type of "Foo" is "type[Foo]"`.
